nixos/modules/tasks/filesystems.nix

```diff
@@ -370,6 +384,10 @@ in
     boot.initrd.systemd.managerEnvironment.SYSTEMD_SYSROOT_FSTAB = initrdFstab;
     boot.initrd.systemd.services.initrd-parse-etc.environment.SYSTEMD_SYSROOT_FSTAB = initrdFstab;
 
+    boot.kernelParams = lib.mkIf config.boot.initrd.systemd.enable [
+      "root=${if config.boot.initrd.gptAutoRoot.enable then "gpt-auto" else "fstab"}"
+    ];
```

Comment:
Since `systemd-gpt-auto-generator` ignores any other value than `root=gpt-auto`, wouldn't it be less invasive to only extend the commandline conditionally here, and not have the `root=fstab` placeholder?

Reply:
I do think we need to be able to do a conditional extend yes. As people might also set `root=/dev/vda3` or `root=dhcpv6` etc

Reply:
`systemd-gpt-auto-generator` attempts to take effect when it sees no `root=` on the cmdline. So if you don't have `root=fstab`, and then you *do* have the root fs defined in `SYSTEMD_SYSROOT_FSTAB` (which is how systemd initrd works right now), then both the gpt generator and the fstab generator will try to make mount units for the root fs and it will break.

Reply:
@ElvishJerricco is this a bug then? The way I understand the manpage, `systemd-gpt-auto-generator` shouldn't try to create mountpoints when they're explcitly defined in fstab. 

> It will also not create mount point configuration for directories which already contain files or if the mount point is explicitly configured in [fstab(5)](https://man7.org/linux/man-pages/man5/fstab.5.html).

Reply:
@nikstur I think that documentation only applies for non-root file systems. Systemd pretty much *expects* you to put `root=` on the cmdline.